src/components/common/navigation/NavigationItem.tsx

```diff
@@ -0,0 +1,44 @@
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import Interaction from '@/components/common/interaction/Interaction';
+
+type SlashPath = `/${string}`;
+
+interface NavigationItemProps {
+  children: string;
+  pathName: SlashPath;
+  disabled?: boolean;
+}
+
+function NavigationItem({ children, pathName, disabled = false }: NavigationItemProps) {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  if (disabled) {
+    return (
+      <button
+        disabled
+        className={`text-object-disabled-dark radius-2xs label-bold-lg px-(--gap-xs) py-(--gap-4xs)`}
+      >
+        {children}
+      </button>
+    );
+  }
+
+  return (
+    <Interaction
+      variant={pathName === location.pathname ? 'brand' : 'default'}
+      density='subtle'
+      isInversed={false}
+    >
+      <button
+        onClick={() => void navigate(pathName)}
```

Comment:
```suggestion
      onClick={handleClick}
```
으로 처리되는 흐름은 어떠신가요? void 를 선언하는 이유가 보통 반환 값을 무시하게 되는데, navigate 에서 이미 부수효과가 발생하기 때문에 void를 명시할 필요가 없다고 생각이 드네요! 

navigate 관련 핸들러 함수를 만들고, (만약에 기존 void의 의도와 비슷하다면) 해당 핸들러를 콜백 처리 해줘도 좋아보입니다!

Reply:
void 작성 안해줘도 동작은 잘 하는데 void 사용하지 않으니까 이런 eslint 에러가 자꾸 뜨더라구요 
Promise가 리턴되는 함수를 void를 리턴해야하는 위치에 작성했을 때 나는 오류라고 해요 🤔

![image](https://github.com/user-attachments/assets/7aa921ba-ad0a-49c6-8181-8391868a04ab)

eslint가 좀 더 엄격해진 거 같아요

Reply:
그쵸 아무래도 이런 문제 때문에 void가 사용되었다고 추측을 했었는데, 실제로 해당 브랜치로 들어가서 확인해보니까 확실히 eslint 쪽 문제인 거 같네요... 조금 생각을 해봤는데 navigate 함수를 Promise로 판단할 수 있는 근거가 크게 없는 거 같은데 혹시 특별한 근거가 보이시나요??!!

navigate 자체는 그냥 이대로 처리해도 좋아보이네요. 별도의 async/await 처리가 콜백 처리 자체도 크게 의미 없어 보이진 않아서 지금처럼 간단하게 표현하는게 제일 나아 보입니다!

Reply:
타입스크립트 문제같네요! navigate 반환 값 타입 자체가 `void | Promise<void>`이라서 그런 거 같아요ㅎㅎ
이대로 가겠습니다!